src/main/kotlin/Mempool.kt

```diff
@@ -1,19 +1,29 @@
-import kotlinx.serialization.Serializable
-import kotlinx.serialization.json.Json
-import java.net.URL
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.serialization.kotlinx.json.*
 
-
-class MempoolSpace {
-    private val json = Json { ignoreUnknownKeys = true }
-
-
-    suspend fun getBlocks(startHeight: Int): BlockResponse {
-        val url = URL("https://mempool.space/api/v1/blocks/$startHeight")
-        val blockJson = url.readText()
-        return json.decodeFromString(BlockResponse.serializer(), blockJson)
+suspend fun getBlockIds(startHeight: Int): List<String> {
+    val client = HttpClient(CIO) {
+        install(ContentNegotiation) {
+            json(Json { ignoreUnknownKeys = true })
+        }
     }
+    val response = client.get("https://mempool.space/api/v1/blocks/$startHeight")
+    val blockIds = Json.decodeFromString<String>(response.toString())
```

Comment:
This isn't the way you should deserialize content. Have you read the documentation for ktor deserialization? As they explain [here](https://ktor.io/docs/serialization-client.html#receive_data) you just need to specify what data structure you want (`BlockId`) and call the `.body()` function and they will handle deserialization, as long as your class matches the JSON you're receiving.

You should *not* convert to string and then deserialize from string, this is highly inefficient.

Reply:
Earlier, I tried doing it the way it it is given in the documentation.
I was facing issue about `response` as unresolved references, and my code was not building. 
However, I'll try to solve the errors by any other method.

Reply:
But when that happens, you need to understand why something didn't work before trying something else. If you didn't understand why the previous code didn't work, you must research it on the internet or try to figure it out. This is important, otherwise you won't be able to write code that you fully master.